server/src/main/java/org/candlepin/model/activationkeys/ActivationKey.java

```diff
@@ -189,6 +186,71 @@ public void setOwner(Owner owner) {
         this.owner = owner;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String toString() {
+        return String.format("ActivationKey [id: %s, name: %s, description: %s]",
+                this.getId(), this.getName(), this.getDescription());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == this) {
+            return true;
+        }
+
+        if (obj instanceof ActivationKey && super.equals(obj)) {
+            ActivationKey that = (ActivationKey) obj;
+
+            // Pull the owner IDs, as we're not interested in verifying that the owners
+            // themselves are equal; just so long as they point to the same owner.
+            String thisOwnerId = this.getOwner() != null ? this.getOwner().getId() : null;
+            String thatOwnerId = that.getOwner() != null ? that.getOwner().getId() : null;
+
+            EqualsBuilder builder = new EqualsBuilder()
+                .append(this.getId(), that.getId())
+                .append(this.getName(), that.getName())
+                .append(this.getDescription(), that.getDescription())
+                .append(thisOwnerId, thatOwnerId)
+                .append(this.getReleaseVer(), that.getReleaseVer())
+                .append(this.getServiceLevel(), that.getServiceLevel())
+                .append(this.isAutoAttach(), that.isAutoAttach())
+                .append(this.getPools(), that.getPools())
+                .append(this.getProducts(), that.getProducts());
```

Comment:
You'll have to be careful here. With Hibernate-backed objects, there's no guarantee we'll get a sane container. As such, with the collections, you may need to use our Util class's ```collectionsAreEqual``` method for comparing them. Take a look at Product.equals for an example.